tests/suites/test_suite_ssl.function

```diff
@@ -5268,3 +5273,79 @@ void conf_group()
     mbedtls_ssl_config_free( &conf );
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_PKCS1_V15:MBEDTLS_SSL_PROTO_TLS1_2:MBEDTLS_RSA_C:MBEDTLS_ECP_DP_SECP256R1_ENABLED:MBEDTLS_SSL_PROTO_DTLS:MBEDTLS_ENTROPY_C:MBEDTLS_RSA_C:MBEDTLS_ECP_DP_SECP384R1_ENABLED:MBEDTLS_CTR_DRBG_C:MBEDTLS_ECP_C */
+void raw_key_agreement_fail( )
+{
+    enum { BUFFSIZE = 17000 };
+    mbedtls_endpoint client, server;
+    mbedtls_psa_stats_t stats;
+
+#if defined(MBEDTLS_TIMING_C)
+    mbedtls_timing_delay_context timer_client, timer_server;
+#endif
+    uint16_t iana_tls_group_list[] = { MBEDTLS_SSL_IANA_TLS_GROUP_SECP256R1,
+                                       MBEDTLS_SSL_IANA_TLS_GROUP_NONE };
+
+    mbedtls_test_message_queue server_queue, client_queue;
+    mbedtls_test_message_socket_context server_context, client_context;
+
+    mbedtls_message_socket_init( &server_context );
+    mbedtls_message_socket_init( &client_context );
+
+    USE_PSA_INIT( );
+
+    /* Client side, force SECP256R1 to make one key bitflip fail
+     * the raw key agreement. Flipping one bit with a Weierstrass
+     * curve (as opposed to a Montgomery curve) has a high chance of
```

Comment:
Unfortunately I am not profficient enough to know how to calculate the probability of this happening. Maybe @mpg, @yanesca, or @gilles-peskine-arm could help?

Reply:
Yes, I also run it several times and didn't catch the failure, but this comment sounds like it is possible. Probably it is very rare case.

Reply:
So if we got format `0x04 || x || y` I think that the test should verify that bit flip is in the first byte (invalid encoding) and bit flip in `x` or `y` (invalid point). Maybe we could check if fake point is on the curve (`y^2 != (y')^2`)? If it is then flip next bit?

Reply:
I think it depends what we want to test: if we just want to ensure that errors from `psa_raw_key_agreement()` are handled (and we don't have any bugs on the error handling path), then any error is good, and just changing the first byte to 0x05 is the simplest and most effective way of achieving that.

Going further and changing coordinates so that they no longer satisfy the curve's equation would only be useful it we wanted to make sure `psa_raw_key_agreement()` fails in all the circumstances it's supposed to. But that should be done in `psa_raw_key_agreement()` unit tests (presumably in `test_suite_psa_crypto`). Here, in `test_suite_ssl`, we're testing the integration of that function into the SSL/TLS module, not the function itself, so any failure is good, and we should use the simplest failure mode.

Reply:
I agree that it depends on what we want to test here. I was thinking that this test is high-level test, so we don't care about the implementation details and treat the software as black box to verify that bit flip can be caught and properly handled. That is why I suggested invalid format and invalid point test cases.

Reply:
Sure, I'll test that on Monday.

Reply:
Updated and verified, that the path now hit is:
```
   if( buf[0] != 0x04 )
            return( MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE );			
```
in `mbedtls_ecp_point_read_binary`.

Reply:
Ok so now we got flipped bit in the first byte which should be always `0x04` and we have verified that `mbedtls_ecp_point_read_binary` fails on checking this value. 
This solves the problem with very rare possibility that the test will fail, but I think that comments now need to by updated. Now we are not flipping bit in point coordinates (x, y):

>  Flipping one bit with a Weierstrass
     * curve (as opposed to a Montgomery curve) has a high chance of ...

I suggest to add comment with information why we picked the first byte.

Reply:
Ok so now we got flipped bit in the first byte which should be always `0x04` and we have verified that `mbedtls_ecp_point_read_binary` fails on checking this value. 
This solves the problem with very rare possibility that the test will fail, but I think that comments now need to by updated. Now we are not flipping bit in point coordinates (x, y):

>  Flipping one bit with a Weierstrass
     * curve (as opposed to a Montgomery curve) has a high chance of ...

I suggest to add comment with information why we picked the first byte.

Reply:
Fixed :)